sqlglot/lineage.py

```diff
@@ -276,8 +276,46 @@ def to_node(
                 reference_node_name=reference_node_name,
                 trim_selects=trim_selects,
             )
+
+        elif scope.pivots:
+            pivot = next(
+                (p for p in scope.pivots if p.alias_or_name == c.table and not p.unpivot), None
+            )
+            if pivot:
+                # The source is a pivot operation, so we need to trace back to the aggregated column
+                pivot_column_mapping = {}
+                pivot_aggs = len(pivot.expressions)
+                columns = pivot.args.get("columns")
+
+                for i, agg in enumerate(pivot.expressions):
+                    agg_cols = list(agg.find_all(exp.Column))
+                    for col in range(i, len(columns), pivot_aggs):
+                        # Go through the field_names in a step manner for this aggfunc
+                        pivot_column_mapping[columns[col].name] = agg_cols
+
+                for agg_column in pivot_column_mapping[c.alias_or_name]:
```

Comment:
I added a whole section where it checks for the implicit column of the pivoted source first. Also added tests for this case.

But in case it is a implicit column, I could not use `c.sql()` because the column `c` belongs to the pivot table `_q_0` and changed the table name manually which I think there is a better way:
https://github.com/step4/sqlglot/blob/cb37b5619acb0e127b75496bef84d24198b5cc70/sqlglot/lineage.py#L307-L314

Reply:
With the refactoring I adapt the column expression now to have the proper table.
in lines 314 - 317